docs/rules/no-unused-expressions.md

```diff
@@ -46,7 +58,11 @@ The following patterns are not considered problems by default:
 ```js
 /*eslint no-unused-expressions: 2*/
 
-{}
+{} // In this context, this is a block statement, not an object literal
+
+{myLabel: anExpression()} // In this context, this is a block statement with a label and expression, not an object literal
+
+function aFunctionDeclaration () {}
```

Comment:
FYI, I've also amended to change the labeled construct to `{myLabel: someVar}` to more clearly show that the construct is considered not a problem, not because of having side effects, but because it is not an expression.

Actually, I wonder whether this strange case is caught by `no-empty` as I think it should. The example there probably ought to also show an empty `{}`.